lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,38 @@
+module RSpec
+  module Core
+    # @private
+    class NoProfiler
+      def example_groups
+        {}
+      end
+    end
+
+    # @private
+    class Profiler
+      Formatters.register self, :example_group_started, :example_group_finished, :example_started
```

Comment:
We already have a profile formatter:

https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/formatters/profile_formatter.rb

Why have 2 profile formatters?  Seems like this stuff could be merged into the existing formatter.